pubtools/_pulp/tasks/garbage_collect.py

```diff
@@ -96,13 +99,29 @@ def clean_all_rpm_content(self):
             LOG.info("No repos found for cleaning.")
             return
         arc_repo = clean_repos[0]
-        deleted_arc = list(arc_repo.remove_content(criteria=criteria))
+
         deleted_content = []
-        for task in deleted_arc:
-            if task.repo_id == "all-rpm-content":
+
+        while True:
+            delete_arc_tasks = list(
+                arc_repo.remove_content(
+                    criteria=criteria, limit=UNASSOCIATE_BATCH_LIMIT
+                )
+            )
+            # The pulp client returns task(s) wrapped in a list
```

Comment:
Still not really sure what this comment means.  I feel like this comment and the surrounding code are making things out to be more complicated than they are.

`remove_content` returns a `Future[list[Task]]`, the Future has already been unpacked at this point implicitly by the `list(...)` in the previous statement, so we just have a `list[Task]` here.  To me saying that it's "wrapped" in a list would indicate that there's a redundant level of list.

If I think about another common list for a moment: command-line arguments are passed as a `list[str]` in `sys.argv`.  Would it be natural to say that the arguments are "wrapped" in a list?  To me it's not, they just natively *are* a list.  If someone wrote code dealing with sys.argv with a comment that arguments were wrapped in a list, I'd think they're trying to call attention to something unusual, like the code having to unexpectedly deal with `list[list[str]]` and having to "unwrap" it to get back to the actual argument list of `list[str]`.

Overall, this comment gives an impression of type confusion.  It might be less confusing if the code around here instead looked like:

```python
delete_arc_tasks = arc_repo.remove_content(
    criteria=criteria, limit=UNASSOCIATE_BATCH_LIMIT
).result()
arc_tasks = [t for t in delete_arc_tasks if t.repo_id == "all-rpm-content"]
for task in arc_tasks:
```

Reply:
A lot of the pulp client functions return a single Task inside a list, which to me felt odd. Though I guess it's by design, in case some functions were intended to return multiple Tasks.

Reply:
Right, the reason for that is to avoid locking the methods into generating precisely one task, as this way they can be extended without breaking backwards-compatibility.

For example, for this particular case it would also have been possible to update `remove_content` to internally do the pagination, rather than requiring the caller to do it.  That would have created one task per "page", and since the return value has always been a list of tasks, it'd be backwards-compatible.  Maybe it's a shame we didn't do it that way actually, I didn't think of that at the beginning.

The same applies to other things like:

- `copy_content` possibly ought to do pagination internally as well
- various methods might want to have side-effects added along the lines of: every time you do X to some repo, it should also automatically be published.